web/src/engine/websites/Bomtoon.ts

```diff
@@ -37,4 +61,53 @@ export default class extends Delitoon {
         return results.distinct();
     }
 
+    public override async FetchPages(chapter: Chapter): Promise<Page<ScrambleParams>[]> {
+        await this.UpdateToken();
+        const url = new URL(`contents/viewer/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.apiUrl);
+        url.searchParams.set('isNotLoginAdult', 'true');
+        const { result, data: { images, isScramble } } = await FetchJSON<APIResult<APIPages>>(this.CreateRequest(url));
+        if (result === 'ERROR') {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+
+        if (isScramble) {
+            const endpoint = new URL(`contents/images/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.apiUrl);
+            const body = JSON.stringify({ line: images[0].line });
```

Comment:
The decryption key for the first image is the same as for all other images?
If not, consider to fetching the keys and decrypt the scramble pattern in the `FetchImage` method.

Reply:
Its the same. As weird as it is, website code does exactly like that.